src/Okta.Sdk/DefaultRetryStrategy.cs

```diff
@@ -0,0 +1,134 @@
+﻿// <copyright file="DefaultRetryStrategy.cs" company="Okta, Inc">
+// Copyright (c) 2014 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Okta.Sdk
+{
+    /// <summary>
+    /// Default retry strategy
+    /// </summary>
+    public class DefaultRetryStrategy : IRetryStrategy
+    {
+        /// <summary>
+        /// The default delta used in the back-off formula to account for some clock skew in our service
+        /// </summary>
+        public static readonly int DefaultBackOffSecondsDelta = 1;
+
+        /// <inheritdoc/>
+        public int MaxRetries { get; set; }
+
+        /// <inheritdoc/>
+        public int RequestTimeOut { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultRetryStrategy"/> class.
+        /// </summary>
+        /// <param name="maxRetries">the number of times to retry</param>
+        /// <param name="requestTimeOut">The request timeout in seconds</param>
+        public DefaultRetryStrategy(int maxRetries, int requestTimeOut)
+        {
+            MaxRetries = maxRetries;
+            RequestTimeOut = requestTimeOut;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultRetryStrategy"/> class.
+        /// </summary>
+        public DefaultRetryStrategy()
+        {
+        }
+
+        /// <inheritdoc/>
+        public async Task<HttpResponseMessage> WaitAndRetryAsync(HttpRequestMessage request, Func<HttpRequestMessage, Task<HttpResponseMessage>> operation)
```

Comment:
Because `DefaultRequestExecutor.SendAsync(...)` accepts a `CancellationToken`, this method also needs to accept a `CancellationToken` and pass it to `CreateDelayAsync` so it can be plumbed through to `Task.Delay`.

Otherwise, when the cancellation token is triggered we won't actually cancel the operation until the current `Task.Delay` completes. It's possible that we'll spend more time in `Task.Delay` than in `operation()` so this is significant.

Reply:
Good call 👍